### PR TITLE
game: add secrets menu, music player and some more pc settings to the debug menu

### PR DESCRIPTION
**Pull request by Hat-Kid**
Adds a secrets menu with PC cheats and a music player to the PC settings debug menu, as well as some missing PC settings like FPS, MSAA and V-Sync.

Also adds some string helper functions to the string library.

### CODE REVIEW

File: goal_src/jak1/engine/debug/default-menu.gc [hunk 1]
```diff
@@ -4585,6 +4687,27 @@
...
+          (flag "OFF" #f (lambda (arg (msg debug-menu-msg)) (when (= msg (debug-menu-msg press)) (set! (-> *pc-settings* gfx-msaa) 1))
+                                                            (case (-> *pc-settings* gfx-msaa) ((1)))))
+          (flag "x2" #f (lambda (arg (msg debug-menu-msg)) (when (= msg (debug-menu-msg press)) (set! (-> *pc-settings* gfx-msaa) 2))
+                                                           (case (-> *pc-settings* gfx-msaa) ((2)))))
+          (flag "x4" #f (lambda (arg (msg debug-menu-msg)) (when (= msg (debug-menu-msg press)) (set! (-> *pc-settings* gfx-msaa) 4))
+                                                           (case (-> *pc-settings* gfx-msaa) ((4)))))
+          (flag "x8" #f (lambda (arg (msg debug-menu-msg)) (when (= msg (debug-menu-msg press)) (set! (-> *pc-settings* gfx-msaa) 8))
+                                                           (case (-> *pc-settings* gfx-msaa) ((8)))))
+          (flag "x16" #f (lambda (arg (msg debug-menu-msg)) (when (= msg (debug-menu-msg press)) (set! (-> *pc-settings* gfx-msaa) 16))
+                                                            (case (-> *pc-settings* gfx-msaa) ((16)))))
```
Comment (ManDude) [inline goal_src/jak1/engine/debug/default-menu.gc:4708]: You can just do `(= (-> *pc-settings* gfx-msaa) 16)` etc here.

File: goal_src/jak1/engine/debug/default-menu.gc [hunk 2]
```diff
@@ -4480,12 +4480,66 @@
...
+                 (lvl (new 'debug 'string 0 (the-as string #f)))
+                 (lvl-menu (new 'debug 'debug-menu ctx level-name))
+                 )
+            (dotimes (j (-> flava-list length))
+                (let ((flava-name (new 'debug 'string 0 (enum->string game-text-id (-> flava-list j))))
+                      (flv (new 'debug 'string 0 (the-as string #f)))
+                      )
+                  (when (and (string= (substring lvl level-name (- (length level-name) 11) (length level-name)) "-level-name") (string= (substring flv flava-name (- (length flava-name) 11) (length flava-name)) "-level-name"))
+                      (copy-string<-data lvl level-name (- (length level-name) 11))
+                      (copy-string<-string (-> lvl-menu name) lvl)
```
Comment (ManDude) [inline goal_src/jak1/engine/debug/default-menu.gc:4516]: Now you can do

```lisp
(let ((str (new 'debug 'string (- (length level-name) 11) #f))
  (set! (-> lvl-menu name) (copyn-string<-string str level-name (- (length level-name) 11)))
  )
```

**Review by ManDude — CHANGES_REQUESTED** [review]
Thanks for adding this!

**Comment by Hat-Kid** [comment]
Figured out a much better way to deal with all the string manip stuff by making use of `*music-list*`.

File: goal_src/jak1/engine/debug/default-menu.gc [hunk 3]
```diff
@@ -4480,12 +4480,59 @@
       )
   )
 
+(defun dm-music-player-func ((lst object))
+  (let ((proc (process-by-name 'dm-player *active-pool*)))
+    (if proc
+        (kill-by-name 'dm-player *active-pool*)
+        (set! *progress-flava* -1)
+      )
+    )
```
Comment (ManDude) [inline goal_src/jak1/engine/debug/default-menu.gc:4489]: You can use the `aif` macro here and save yourself the let. You can also just omit the check entirely, it's fine for `kill-by-name` not to find anything.

File: goal_src/jak1/kernel/gstring.gc [hunk 4]
```diff
@@ -47,6 +47,42 @@
   dst
   )
 
+(defun copyn-string<-string ((dst string) (src string) (len int))
+  "Copy len bytes of data from one string to another, like strcpy"
```
Comment (ManDude) [inline goal_src/jak1/kernel/gstring.gc:51]: `strncpy`

**Review by ManDude — CHANGES_REQUESTED** [review]

**Review by ManDude — APPROVED** [review]
seems good to me right now!

**Pull request merged**